src/stdlib_experimental_io.f90

```diff
@@ -300,51 +300,53 @@ integer function open(filename, mode) result(u)
 
 mode_ = parse_mode(optval(mode, ""))
 
-if (mode_(1:2) == 'r ') then
+select case (mode_(1:2))
```

Comment:
It looks cleaner with `select case`.
However, I usually avoid to use `select case` with `character` since I got memory issues (leaks) with Intel Fortran compiler. Any of you observed such issues?

Reply:
If there are leaks, then we'll revert it.